test/jdk/java/net/httpclient/HttpResponseLimitingTest.java

```diff
@@ -0,0 +1,396 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8328919
+ * @summary tests `limiting()` in `HttpResponse.Body{Handlers,Subscribers}`
+ * @key randomness
+ * @library /test/lib
+ *          /test/jdk/java/net/httpclient/lib
+ * @build jdk.httpclient.test.lib.common.HttpServerAdapters
+ *        jdk.test.lib.RandomFactory
+ *        jdk.test.lib.net.SimpleSSLContext
+ * @run junit HttpResponseLimitingTest
+ */
+
+import jdk.httpclient.test.lib.common.HttpServerAdapters.HttpTestServer;
+import jdk.test.lib.RandomFactory;
+import jdk.test.lib.net.SimpleSSLContext;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import javax.net.ssl.SSLContext;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpHeaders;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandler;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.net.http.HttpResponse.BodySubscriber;
+import java.net.http.HttpResponse.BodySubscribers;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Flow.Subscription;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.net.http.HttpClient.Builder.NO_PROXY;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Arrays.copyOfRange;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class HttpResponseLimitingTest {
+
+    private static final Random RANDOM = RandomFactory.getRandom();
+
+    private static final byte[] RESPONSE_BODY = "random non-empty body".getBytes(UTF_8);
+
+    private static final String RESPONSE_HEADER_NAME = "X-Excessive-Data";
+
+    /**
+     * A header value larger than {@link #RESPONSE_BODY} to verify that {@code limiting()} doesn't affect header parsing.
+     */
+    private static final String RESPONSE_HEADER_VALUE = "!".repeat(RESPONSE_BODY.length);
```

Comment:
Did you mean this to be:
```java
"!".repeat(RESPONSE_BODY.length + 1);
```
`"x".repeat(1)` will generate a String of length 1.

In the context of what we are doing in this test, it won't matter though.

Reply:
No, it wasn't an oversight, since `data = header name + header value` which is larger than `RESPONSE_BODY`. Nevertheless, apparently it is not clear to the reader. b75938265560ae7c62e88e60a3895e37889316ad replaces it with `RESPONSE_BODY.length + 1` as you suggested.